rep-2006.rst

```diff
@@ -0,0 +1,71 @@
+REP: 2006
+Title: ROS 2 Vulnerability Disclosure Policy
+Author: ROS 2 Security Working Group, Chris Lalancette <clalancette@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 12-May-2020
+Post-History:
+
+
+Abstract
+========
+
+This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.
```

Comment:
Yeah, the scope of this was one of my concerns as well.  We can't possibly be responsible for evaluating and fixing security bugs in the millions of lines of code covered by REP-2005.

That being said, the main intention here is to give us time to react before disclosure.  For most of the heavily-used packages, I would expect that the maintainers would be responsive and work to fix the issue.  If there are packages that have gone inactive, and the developers don't respond, there's not too much we can do about it.  Maybe at that point the response is to delist the package from REP-2005.

Your point about how we contact the maintainers and how they should respond is a good one.  I don't think it is reasonable to expect that every project setup a `security@` closed mailing list.  I think the best we can do in general is to directly contact the maintainers listed in the package.xml (though I'm open to other suggestions as well).  I'll add some verbiage around that to start with.

Reply:
>  I'll add some verbiage around that to start with.

Actually, though, I don't think this is the right place for that.  This REP is a statement from the developers of ROS to the wider world, not a documentation of the internal processes.  If we are going to add verbiage, I would think it would be in REP-2005.  I'll add something there instead.

Reply:
> Actually, though, I don't think this is the right place for that. This REP is a statement from the developers of ROS to the wider world, not a documentation of the internal processes. 

I agree. There are two aspects to handling vuln reports: the process we go through with the reporter, and the internal process of actually triaging and fixing the vuln. This is the former, and I think it's important for it to stay that way. The latter still needs to be written, and I'm not sure it belongs in REP-2005 either. Maybe the developer guide?

Reply:
Yeah I agree it makes sense to have it elsewhere. Could be a process REP or something less formal if we think it requires iterations / maturing. This comment is mostly pointing out that this was a pending question when we drafted this document and that this REP does imply a process that is currently not defined anywhere. It would need to be for this REP to take effect

Reply:
That kind of internal process documentation seem to me like it would go in a developer guide or a security group working document.  It doesn't exactly seem like an REP.  I would suggest a new page on https://github.com/ros2/ros2_documentation, but other places are fine as well.

Given that I have no familiarity with what these processes are expected to be, would someone be willing to step up and write down some of what has been discussed as the internal process for handling security reports?

Reply:
This and a handful of open issues raised in the discussions (e.g., how to handle embargoed issues, how deep in the dependency stack we travel, providing training materials, providing VDP tempates to vendors). I'll gather these from the [WG Wiki Page](http://wiki.ros.org/action/fullsearch/ROS2/WorkingGroups/Security) and add them to the working agenda.

> It would need to be for this REP to take effect

I'd like to move this forward as we continue to work through the supporting processes, similar to how a VDP was required with REP 2005 before this doc was issued.

Reply:
Excellent, thanks for taking the lead here @SidFaber.  It's appreciated.

Reply:
Given the above discussion, should I roll back the changes on REP-2005?

Reply:
> I'd like to move this forward as we continue to work through the supporting processes, similar to how a VDP was required with REP 2005 before this doc was issued.

Yeah, that works fine for me.  I don't think having those processes in place is a prerequisite for merging this.

Reply:
I'm going to resolve this conversation, as there is no further action to be taken on this PR for this.  We'll discuss the ongoing internal processes in the Security Working Group meeting.  Feel free to unresolve and/or add another comment if you feel differently.